### PR TITLE
fix(cli): ensure that pre- and post-test output is flushed at the appropriate times

### PR DESCRIPTION
**Pull request by mmastrac**
Some `deno_std` tests were failing to print output that was resolved after the last test finished. In addition, output printed before tests began would sometimes appear above the "running X tests ..." line, and sometimes below it depending on timing.

We now guarantee that all output is flushed before and after tests run, making the output consistent.

Pre-test and post-test output are captured in `------ pre-test output ------` and `------ post-test output ------` blocks to differentiate them from the regular output blocks.

Here's an example of a test (that is much noisier than normal, but an example of what the output will look like):

```
Check ./load_unload.ts
------- pre-test output -------
load
----- output end -----
running 1 test from ./load_unload.ts
test ...
------- output -------
test
----- output end -----
test ... ok ([WILDCARD])
------- post-test output -------
unload
----- output end -----
```

### CODE REVIEW

File: tests/integration/run_tests.rs [hunk 1]
```diff
@@ -4560,15 +4560,14 @@ async fn websocket_server_multi_field_connection_header() {
   assert!(child.wait().unwrap().success());
 }
 
-// TODO(bartlomieju): this should use `deno run`, not `deno test`; but the
-// test hangs then. https://github.com/denoland/deno/issues/14283
 #[tokio::test]
 async fn websocket_server_idletimeout() {
+  test_util::timeout!(60);
```
Comment (mmastrac) [inline tests/integration/run_tests.rs:4565]: This test was using `deno test` to bypass a hang, but that stopped working with this change. I (mostly) fixed the test by just forcing an exit.

File: tests/testdata/test/load_unload.out [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+----- output end -----
+running 1 test from [WILDCARD]/load_unload.ts
+test ...
+------- output -------
+test
+----- output end -----
 test ... ok ([WILDCARD])
+------- post-test output -------
+unload
+----- output end -----
```
Comment (bartlomieju) [inline tests/testdata/test/load_unload.out:13]: This is a bit confusing - we have `output end` two end. Is this actually expected?

Comment (mmastrac) [reply tests/testdata/test/load_unload.out:13]: The `output end` matches with `post-test output`. 

Maybe I can clarify this by making the pairs `------- post-test output -------` and `------- post-test output end -------` instead. WDYT?

Comment (bartlomieju) [reply tests/testdata/test/load_unload.out:13]: I think it will be less confusing, so +1 from me

**Pull request merged**